master/buildbot/status/web/build.py

```diff
@@ -462,6 +319,152 @@ def getChild(self, path, req):
+        context['build_reason'] = self.build_status.getReason()
+        context['build_is_resuming'] = self.build_status.isResuming()
+        context['build_is_finished'] = self.build_status.isFinished()
+
+        context['path_to_builder'] = path_to_builder(request, self.build_status.getBuilder())
+        context['path_to_builders'] = path_to_builders(request, builder.getProject())
+        context['path_to_codebases'] = path_to_codebases(request, builder.getProject())
+        context['build_url'] = path_to_build(request, self.build_status, False)
+        context['slave_debug_url'] = self.getBuildmaster(request).config.slave_debug_url
+        context['codebases_arg'] = getCodebasesArg(request=request)
+        context['parent_build_url'] = None
+        context['top_build_url'] = None
+
+        start, end = self.build_status.getTimes()
+        context['start'] = time.ctime(start)
+        context['end'] = time.ctime(end) if end else None
+        if not end:
+            end = util.now()
+        context['elapsed'] = util.formatInterval(end - start)
+
+        context['authz'] = self.getAuthz(request)
+        context['has_changes'] = False
+        context['tests_link'] = None
+        context['resume'] = None
+        context['top_build_name'] = None
+        context['parent_build_name'] = None
+        context['result_css'] = ""
+        context['slave_url'] = ""
+        context['steps'] = []
+        context['properties'] = []
+
+        return context
+
+    def __get_force_scheduler_parameters(self, request):
+        """ This method return dictionary of parameters from scheduler
+
+        :param request: http request object
+        :return: dictionary with parameters
+        """
+        scheduler_name = self.build_status.getProperty("scheduler", None)
+        parameters = {}
+        scheduler = next(
+            ifilter(
+                lambda s: s.name == scheduler_name and isinstance(s, ForceScheduler),
+                self.getBuildmaster(request).allSchedulers(),
+            ),
+            None
+        )
+        if not scheduler:
+            return {}
+
+        for p in scheduler.all_fields:
+            parameters[p.name] = p
+
+        return parameters
+
+    def __get_properties(self, parameters):
+        """ This method prepare properties of build
+        TODO refactor
+
+        :param parameters: dictionary with parameter object
+        :return: list of properties object
+        """
+        properties = []
+        for name, value, source in self.build_status.getProperties().asList():
+            if not isinstance(value, dict):
+                cxt_value = unicode(value)
+            else:
+                cxt_value = value
+
+            if name == 'submittedTime':
+                cxt_value = time.ctime(value)
+
+            prop = {
+                'name': name,
+                'value': cxt_value,
+                'short_value': cxt_value[:500],
```

Comment:
I figure out that we cannot slice in this way because cxt_value sometimes is a dict!
`if len(cxt_value) > 500` helps for this situation because dict does not have more that 500 keys.

I don't know if author prepared this `if` exactly for case with dict, but checking type by checking length of checked element is horrible. Of course, if we put again `if` and dict will have more than 500 keys the Http 500 error page will occur as well